sympy/series/fourier.py

```diff
@@ -440,16 +438,213 @@ def __sub__(self, other):
         return self.__add__(-other)
 
 
-class FiniteFourierSeries(Basic):
-    def __new__(cls, *args):
-        obj = Basic.__new__(cls, *args)
-        return obj
+class FiniteFourierSeries(FourierSeries):
+    r"""Represents Finite Fourier sine/cosine series.
 
-    def truncate(self, n=3):
-        return Add(*self._args)
+       For how to compute Fourier series, see the :func:`fourier_series`
+       docstring.
+
+       Parameters
+       ==========
+       f : Expr
+            Expression for finding fourier_series
+
+       limits : ( x, start, stop)
+            x is the independent variable for the expression f
+            (start, stop) is the period of the fourier series
+
+       exprs: (a0, an, bn) or Expr
+            a0 is the constant term a0 of the fourier series
+            an is a dictionary of coefficients of cos terms
+             `an[k] = coefficient of cos(pi*(k/L)*x)`
+            bn is a dictionary of coefficients of sin terms
+             `bn[k] = coefficient of sin(pi*(k/L)*x)`
+
+            or exprs can be an expression to be converted to fourier form
+
+       Methods
+       =======
+       This class is an extension of FourierSeries class.
+       Please refer to sympy.series.fourier.FourierSeries for
+       further information.
+
+       See Also
+       ========
+
+       sympy.series.fourier.FourierSeries
+       sympy.series.fourier.fourier_series
+       """
+
+    def __new__(cls, f, limits, exprs):
+
+        if not (type(exprs) == tuple and len(exprs) == 3):  # exprs is not of form (a0, an, bn)
+
+            # Converts the expression to fourier form
+            exprs = exprs.as_coeff_add()
```

Comment:
How about having something like `c, e` on the left hand side.

Reply:
Yeah, it would be more readable but I didn't want to add unnecessary variables. I don't think it would be necessary.

Reply:
I think that the bytecode will be simpler with a local variable. Using a few more slots on the local parameter stack should not be problem.

Reply:
I didn't do because this eventually is a single statement and the variables used are not going far. I can change that if you suggest.